### PR TITLE
[APIMANAGER-4468] Throw RuntimeException instead of Error for invalid Json

### PR DESCRIPTION
**Pull request by manoramahp**
This fix is to change throwing RuntimeException instead of Error when received an invalid Json payload, as Synapse requires exception to be thrown
Related JIRA : https://wso2.org/jira/browse/APIMANAGER-4468

### CODE REVIEW

File: modules/commons/src/test/java/org/apache/synapse/commons/json/JsonStreamingBuilderTest.java [hunk 1]
```diff
@@ -19,8 +19,36 @@
 package org.apache.synapse.commons.json;
 
 import junit.framework.TestCase;
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.builder.Builder;
+import org.apache.axis2.context.MessageContext;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import java.io.InputStream;
```
Comment (milindaperera) [inline modules/commons/src/test/java/org/apache/synapse/commons/json/JsonStreamingBuilderTest.java]: Shall we add little description about the test case even though this is an old empty test-case

File: modules/commons/src/test/java/org/apache/synapse/commons/json/JsonStreamingBuilderTest.java [hunk 2]
```diff
@@ -19,8 +19,36 @@
...
+
+    public  void testInvalidJson(String jsonIn) {
+        try {
+            MessageContext message = Util.newMessageContext();
+            InputStream inputStream = Util.newInputStream(jsonIn.getBytes());
+            OMElement element  = jsonBuilder.processDocument(inputStream, "application/json", message);
+            message.getEnvelope().getBody().addChild(element);
+            log.info(message.getEnvelope().getBody().toString());
+        } catch (Exception e) {
+            assertEquals(e.getMessage(),"Error: could not match input");
```
Comment (milindaperera) [inline modules/commons/src/test/java/org/apache/synapse/commons/json/JsonStreamingBuilderTest.java]: are we testing the exception here? no asserts in try section

Comment (manoramahp) [inline modules/commons/src/test/java/org/apache/synapse/commons/json/JsonStreamingBuilderTest.java]: yes, we need to test whether we're getting an exception

Comment (milindaperera) [inline modules/commons/src/test/java/org/apache/synapse/commons/json/JsonStreamingBuilderTest.java]: Are we testing whether the runtime exception is thrown correctly? if so we can check instance of RuntimeException

**Comment by manoramahp** [comment]
Opened https://github.com/wso2/wso2-synapse/pull/818 with the suggested changes
